src/v0/destinations/mp/transform.js

```diff
@@ -379,20 +385,111 @@ const processSingleMessage = async (message, destination) => {
       return processAliasEvents(message, message.type, destination);
     case EventType.GROUP:
       return processGroupEvents(clonedMessage, clonedMessage.type, destination);
-
     default:
       throw new InstrumentationError(`Event type ${clonedMessage.type} is not supported`);
   }
 };
 
 const process = async (event) => processSingleMessage(event.message, event.destination);
 
+const processEvents = async (inputs, reqMetadata) =>
+  await Promise.all(
+    inputs.map(async (event) => {
+      try {
+        if (event.message.statusCode) {
+          // already transformed event
+          return { output: event };
+        }
+
+        // if not transformed
+        return {
+          output: {
+            message: await process(event),
+            metadata: event.metadata,
+            destination: event.destination,
+          },
+        };
+      } catch (error) {
+        const errRespEvent = handleRtTfSingleEventError(event, error, reqMetadata);
+        return { error: errRespEvent };
+      }
+    }),
+  );
+
```

Comment:
@Gauravudia can you plan unit tests for `batchEvents` & `processAndChunkEvents`, `generateBatchedPayloadForArray` functions in a separate PR ?